### PR TITLE
`r\linux_virtual_machine_scale_set` `r\windows_virtual_machine_scale_set`: Rename `terminate_notification` to `termination_notification`

### PR DESCRIPTION
**Pull request by myc2h6o**
Updated per comment
\--------
Updating the property name to the correct word and match the one in `r\orchestrated_virtual_machine_scale_set`.
https://github.com/hashicorp/terraform-provider-azurerm/blob/f232d0a1ba5ef5356db2b703387bb11aec2ea691/internal/services/compute/orchestrated_virtual_machine_scale_set_resource.go#L193

This property is `Optional` + `Computed`, so in the `Update` function, by simply checking the `HasChange` for each of them, it can handle the migration from old property name to new one (If property name changes with value remaining same, no changes are detected).

### CODE REVIEW

**Comment by myc2h6o** [comment]
Test result:
$ TF_ACC=1 go test -v ./internal/services/compute -run='TestAccOrchestratedVirtualMachineScaleSet_Te
=== RUN   TestAccOrchestratedVirtualMachineScaleSet_TerminateNotification
=== PAUSE TestAccOrchestratedVirtualMachineScaleSet_TerminateNotification
=== RUN   TestAccOrchestratedVirtualMachineScaleSet_TerminationNotificationMigration
=== PAUSE TestAccOrchestratedVirtualMachineScaleSet_TerminationNotificationMigration
=== CONT  TestAccOrchestratedVirtualMachineScaleSet_TerminateNotification
=== CONT  TestAccOrchestratedVirtualMachineScaleSet_TerminationNotificationMigration
--- PASS: TestAccOrchestratedVirtualMachineScaleSet_TerminateNotification (580.10s)
--- PASS: TestAccOrchestratedVirtualMachineScaleSet_TerminationNotificationMigration (619.60s)
PASS
ok      github.com/hashicorp/terraform-provider-azurerm/internal/services/compute       620.328s

**Review by katbyte — CHANGES_REQUESTED** [review]
@myc2h6o - could we instead rename terminate to termination as from the docs this is a "termination" notification and that is the correct word, and is more explicit on what this does.

thanks

**Comment by myc2h6o** [comment]
New test result after the update:
image

**Comment by myc2h6o** [comment]
Hi @katbyte Thanks for the comment, yes `termination_notification` seems more reasonable. I've updated the pr, please take a look

**Comment by myc2h6o** [comment]
Test result after update:
image

**Comment by myc2h6o** [comment]
Updated test result
image

**Review by katbyte — CHANGES_REQUESTED** [review]

File: internal/services/compute/virtual_machine_scale_set.go [hunk 1]
```diff
@@ -1398,7 +1398,33 @@ func FlattenVirtualMachineScaleSetRollingUpgradePolicy(input *compute.RollingUpg
...
 }
 
+// TODO remove VirtualMachineScaleSetTerminateNotificationSchema in 4.0
 func VirtualMachineScaleSetTerminateNotificationSchema() *pluginsdk.Schema {
+	return &pluginsdk.Schema{
+		Type:       pluginsdk.TypeList,
+		Optional:   true,
+		Computed:   true,
+		MaxItems:   1,
+		Deprecated: "`terminate_notification` has been renamed to `termination_notification` and will be removed in 4.0.",
```
Comment (myc2h6o) [inline internal/services/compute/virtual_machine_scale_set.go:1408]: Deprecation notice is put here

File: internal/services/compute/linux_virtual_machine_scale_set_resource.go [hunk 2]
```diff
@@ -1210,3 +978,267 @@ func resourceLinuxVirtualMachineScaleSetDelete(d *pluginsdk.ResourceData, meta i
...
+
+		// Computed
+		"unique_id": {
+			Type:     pluginsdk.TypeString,
+			Computed: true,
+		},
+	}
+
+	if !features.FourPointOhBeta() {
+		out["terminate_notification"] = VirtualMachineScaleSetTerminateNotificationSchema()
```
[the remark this answers is not in the document]
Comment (myc2h6o) [reply internal/services/compute/linux_virtual_machine_scale_set_resource.go:1239]: Yeah, it's added in `VirtualMachineScaleSetTerminateNotificationSchema` in virtual_machine_scale_set.go: https://github.com/hashicorp/terraform-provider-azurerm/blob/dd5af47fb37f86cb5511fe6e274dce66071f75b9/internal/services/compute/virtual_machine_scale_set.go#L1408

[the remark this answers is not in the document]
Comment (katbyte) [reply internal/services/compute/linux_virtual_machine_scale_set_resource.go:1239]: ah perfect! thanks :)

**Review by katbyte — APPROVED** [review]
Thanks @myc2h6o - LGTM 🚀

**Pull request merged**